vcd/resource_vcd_network_direct.go

```diff
@@ -95,10 +112,103 @@ func resourceVcdNetworkDirectCreate(d *schema.ResourceData, meta interface{}) er
 
 	err = vdc.CreateOrgVDCNetworkWait(orgVDCNetwork)
 	if err != nil {
-		return fmt.Errorf("error: %#v", err)
+		return fmt.Errorf("error: %s", err)
+	}
+
+	network, err := vdc.GetOrgVdcNetworkByName(networkName, true)
+	if err != nil {
+		return fmt.Errorf("error retrieving network %s after creation", networkName)
+	}
+	d.SetId(network.OrgVDCNetwork.ID)
+	return resourceVcdNetworkDirectRead(d, meta)
+}
+
+func resourceVcdNetworkDirectRead(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		return fmt.Errorf("[network direct read] "+errorRetrievingOrgAndVdc, err)
+	}
+
+	identifier := d.Id()
+
+	if identifier == "" {
+		identifier = d.Get("name").(string)
+	}
+	network, err := vdc.GetOrgVdcNetworkByNameOrId(identifier, false)
+	if err != nil {
+		log.Printf("[DEBUG] Network %s no longer exists. Removing from tfstate", identifier)
+		return fmt.Errorf("[network direct read] network %s not found ", network.OrgVDCNetwork.Name)
 	}
 
-	d.SetId(d.Get("name").(string))
+	_ = d.Set("name", network.OrgVDCNetwork.Name)
+	_ = d.Set("href", network.OrgVDCNetwork.HREF)
+	_ = d.Set("shared", network.OrgVDCNetwork.IsShared)
+
+	parentNetwork := network.OrgVDCNetwork.Configuration.ParentNetwork
+	if parentNetwork == nil {
+		return fmt.Errorf("[network direct read] no parent network found for %s", network.OrgVDCNetwork.Name)
+	}
+	_ = d.Set("external_network", parentNetwork.Name)
+
+	externalNetwork, err := vcdClient.GetExternalNetworkByName(parentNetwork.Name)
+	if err != nil {
+		return fmt.Errorf("[network direct read] error fetching external network %s ", parentNetwork.Name)
+	}
+
+	enConf := externalNetwork.ExternalNetwork.Configuration
+	if enConf == nil || enConf.IPScopes == nil || len(enConf.IPScopes.IPScope) == 0 {
+		return fmt.Errorf("[network direct read] error retrieving details from external network %s", externalNetwork.ExternalNetwork.Name)
+	}
+	_ = d.Set("external_network_gateway", externalNetwork.ExternalNetwork.Configuration.IPScopes.IPScope[0].Gateway)
```

Comment:
Only one IpScope possible?

Reply:
Yes. This is the gateway from the external network. Unlike the edge gateway, which can connect to several external networks, the direct network can only have one